src/biokbase/narrative/app_util.py

```diff
@@ -874,34 +884,43 @@ def transform_object_value(transform_type: Optional[str], value: Optional[str])
 
     if is_upa and transform_type in ["upa", "resolved-ref"]:
         return value
-    if is_ref and not is_upa and transform_type in ["ref", "unresolved-ref"]:
+    if (
+        not is_upa
+        and is_ref
+        and transform_type in ["ref", "unresolved-ref", "putative-ref"]
+    ):
         return value
     if not is_upa and not is_ref and transform_type is None:
         return value
 
-
     search_ref = value
     if not is_upa and not is_ref:
         search_ref = f"{system_variable('workspace')}/{value}"
     try:
         obj_info = clients.get("workspace").get_object_info3(
             {"objects": [{"ref": search_ref}]}
         )
-    except Exception:
+    except Exception as e:
+        # a putative-ref can be an extant or a to-be-created object; if the object
+        # is not found, the workspace name/object name can be returned
+        if transform_type == "putative-ref" and (
+            "No object with name" in str(e) or "No object with id" in str(e)
+        ):
+            return search_ref
```

Comment:
very important part! If it's a putative ref, it's OK if the object doesn't (yet) exist.

Reply:
The only part this could get weird is if the input value was an UPA, and the user's expecting a ref back. Then they'll just get the UPA back.

Which, admittedly, is very silly submitting an UPA and wanting a putative-ref, and in the real world would (hopefully) never be part of a design.